ts/src/connections/tunnelClientBase.ts

```diff
@@ -236,6 +250,36 @@ export class TunnelClientBase
         if (this.connectionProtocol === webSocketSubProtocolv2) {
             pfs.messageFactory = this;
             pfs.onForwardedPortConnecting((e) => this.onForwardedPortConnecting(e));
+            pfs.remoteForwardedPorts.onPortAdded((e) => this.onForwardedPortAdded(pfs, e));
+            pfs.remoteForwardedPorts.onPortUpdated((e) => this.onForwardedPortAdded(pfs, e));
+        }
+    }
+
+    private onForwardedPortAdded(pfs: PortForwardingService, e: ForwardedPortEventArgs) {
+        const port = e.port.remotePort;
+        if (typeof port !== 'number') {
+            return;
+        }
+
+        // If there are disconnected streams for the port, re-connect them now.
+        const disconnectedStreamsCount = this.disconnectedStreams.get(port)?.length ?? 0;
+        for (let i = 0; i < disconnectedStreamsCount; i++) {
+            pfs.connectToForwardedPort(port).catch((error) => {
+                this.trace(
+                    TraceLevel.Warning,
+                    0,
+                    `Failed to reconnect to forwarded port ${port}: ${error}`);
+
+                // The host is no longer accepting connections on the forwarded port?
+                // Clear the list of disconnected streams for the port, because
+                // it seems it is no longer possible to reconnect them.
+                const streams = this.disconnectedStreams.get(port);
+                if (streams) {
+                    while (streams.length > 0) {
+                        streams.pop()!.dispose();
+                    }
+                }
+            });
```

Comment:
I noticed the C# implementation of `onForwardedPortAdded` had a success message `"Reconnected stream to forwarded port {port}"`. Should we add that in a `.then()` here too?

Reply:
Added.